rust/processor/src/processors/fungible_asset_processor.rs

```diff
@@ -415,10 +458,63 @@ impl ProcessorTrait for FungibleAssetProcessor {
     }
 }
 
+/// Gets coin to fungible asset mappings from transactions by looking at CoinInfo
+/// This is very similar code to part of parse_v2_coin
+pub async fn get_coin_to_fa_mapping(transactions: &[Transaction]) -> CoinToFungibleAssetMappings {
+    // First collect all metadata from transactions
+    let data: Vec<_> = transactions
+        .par_iter()
+        .map(|txn| {
+            let mut kv_mapping: CoinToFungibleAssetMappings = AHashMap::new();
+
+            let txn_version = txn.version as i64;
+            let transaction_info = txn.info.as_ref().expect("Transaction info doesn't exist!");
+            for (index, wsc) in transaction_info.changes.iter().enumerate() {
+                if let Change::WriteResource(wr) = wsc.change.as_ref().unwrap() {
+                    if let Some(fa_metadata) =
+                        RawFungibleAssetMetadataModel::get_v1_from_write_resource(
+                            wr,
+                            index as i64,
+                            txn_version,
+                            NaiveDateTime::default(), // placeholder
+                        )
+                        .unwrap_or_else(|e| {
+                            tracing::error!(
+                                transaction_version = txn_version,
+                                index = index,
+                                error = ?e,
+                                "[Parser] error parsing fungible metadata v1");
+                            panic!("[Parser] error parsing fungible metadata v1");
+                        })
+                    {
+                        let coin_to_fa_mapping =
+                            RawCoinToFungibleAssetMapping::from_raw_fungible_asset_metadata(
+                                &fa_metadata,
+                            );
+                        kv_mapping.insert(
+                            coin_to_fa_mapping.coin_type.clone(),
+                            coin_to_fa_mapping.fungible_asset_metadata_address.clone(),
+                        );
+                    }
+                }
+            }
+            kv_mapping
+        })
+        .collect();
+    let mut kv_mapping: CoinToFungibleAssetMappings = AHashMap::new();
+    for mapping in data {
+        kv_mapping.extend(mapping);
+    }
```

Comment:
Can try .into_par_iter().fold() https://docs.rs/rayon/latest/rayon/iter/struct.Fold.html

Reply:
hmm this is also the existing paradigm let's fix it together in a different PR lol.